### ISSUE TITLE
Flags for Tor

### ISSUE THREAD

**Issue by dodedov** [tutao/tutanota#2606]
Windows 10
v3.80.2
Tor 0.4.3.6 (as service)

For the Tutanota Desktop label, I add flags:
`--proxy-server="socks5://127.0.0.1:9050" --host-resolver-rules="MAP * ~NOTFOUND , EXCLUDE 127.0.0.1"`

An error that can't connect to the server. These same flags add to GoogleChrome - everything works. Other Electron based apps also work with these flags.

**Comment by charlag** [comment]
Can you please make sure that's it's the same exit node (when you test browser vs app) somehow?

**Comment by dodedov** [comment]
> Can you please make sure that's it's the same exit node (when you test browser vs app) somehow?

Yes. The IP test in the browser shows not my IP address. In the Element app, the last entry is not from my IP address.
p.s. In my country, Tutanota is blocked, so you have to use Tor for Tutanota Desktop.

**Comment by charlag** [comment]
Can you open inspector with F12 and check where does it send requests?

**Comment by dodedov** [comment]
> Can you open inspector with F12 and check where does it send requests?

Please explain more about what to do. I don't use a Chrome inspector.

**Comment by charlag** [comment]
press F12, go to network tab, clock login, you should see requests, Click one of them and check request URL.
I'm not sure that it will show the proxy there but at least we could see what's going on

**Comment by dodedov** [comment]
> press F12, go to network tab, clock login, you should see requests, Click one of them and check request URL.
> I'm not sure that it will show the proxy there but at least we could see what's going on

`failed to request GET https://mail.tutanota.com/rest/sys/saltservice?_body=%7B%22_format%22%3A%220%22%2C%22mailAddress%22%3A%22dodedov%40tutanota.com%22%7D`

`error ConnectionError {name: "ConnectionError", message: "0:  | GET /rest/sys/saltservice", data: undefined, stack: "ConnectionError: 0:  | GET /rest/sys/saltservice↵ …%20Desktop/resources/app.asar/worker.js:1:700497)", isOperational: true}`

`GET https://tutanota.com/images/maintenancecheck.png net::ERR_CONNECTION_RESET`

**Comment by charlag** [comment]
Thanks
I tried to find more info on proxy in Electron but could not do it yet. I think it might be that it's just the relay that it picks is temporarily blocked, we don't have to do anything special I hope.
Otherwise it might be related to show appImage invokes the app but I would assume it passes all arguments through.

**Comment by dodedov** [comment]
What else can I do? Is there another way to make Tutanota Desktop work by bypassing the country's blocking?

p.s. Excuse me, I'm writing requests for functionality here, are there any plans to add a proxy setting before entering the login and password in the future? Not to enter flags.

**Comment by charlag** [comment]
Do you have any control over which exit node it picks? If it can pick another one, I would try that
We do not have plans yet. We might add something to avoid blocks later on.

**Comment by dodedov** [comment]
I figured out what the problem was. The Tutanota process has stalled. That's why all the new flags didn't work.
I confirm that the flags are working to work through Tor and bypass the locks from the country.
`--proxy-server="socks5://127.0.0.1:9050" --host-resolver-rules="MAP * ~NOTFOUND , EXCLUDE 127.0.0.1"`

Sorry to trouble you!

**Comment by charlag** [comment]
Yes, it works in background if flag is set in settings.
Glad to know you have it working, I will close the issue

**Comment by jasonbrown1965** [comment]
#eli5 fix info request, newb, plz

- restart app, or ?
- wait for it to _unstall_ the install ?
